pre_commit_hooks/check_vcs_permalinks.py

```diff
@@ -8,7 +8,7 @@
 
 
 GITHUB_NON_PERMALINK = re.compile(
-    b'https://github.com/[^/]+/[^/]+/blob/master/[^# ]+#L\d+',
+    b'https?://(www\.)?github.com/[^/]+/[^/]+/blob/master/[^# ]+#',
```

Comment:
github redirects `http` and `www.github.com` to `https://github.com`, so I don't think this is worth covering / complicating

Reply:
Yes, but if you check in `http://github.com/asottile/test/blob/master/foo#L1` into a codebase, then you would check in a link that could break in the future. The regex before my change doesn't catch that case

Reply:
where would you get such a link? manually typed out?

Reply:
That said, this is by far the less controversial part of this patch -- split out the other part and I can accept this

Reply:
Hmm good point. Will think about this. Thinking more, you could drop the `https?://(www\.)?` part entirely in the regex (it would transparently catch the same scenarios)

Reply:
Hmm OTOH if someone has a line length limit they might drop the `https://` 🤔